src/main/java/org/gentoo/java/ebuilder/portage/MavenVersion.java

```diff
@@ -14,7 +14,7 @@
      * Pattern for parsing maven version number.
      */
     private static final Pattern PATTERN_VERSION = Pattern.compile(
-            "^(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:-(.*))?$");
+            "^(\\d+)(?:\\.(\\d+))?(?:[\\.b](\\d+))?(?:[\\.-](.*))?$");
```

Comment:
trying to figure out whether these version match such a versioning scheme. i found both examples are really used:
https://mvnrepository.com/artifact/org.jboss.logging/jboss-logging
https://mvnrepository.com/artifact/org.apache-extras.beanshell/bsh

here are maven version rules described:
http://www.mojohaus.org/versions-maven-plugin/version-rules.html

> The current implementation of DefaultArtifactVersion in the core of Maven expects that version numbers will have a very specific format:
> 
> `<MajorVersion [> . <MinorVersion [> . <IncrementalVersion ] ] [> - <BuildNumber | Qualifier ]>`
> 
> Where MajorVersion, MinorVersion, IncrementalVersion and BuildNumber are all numeric and Qualifier is a string. If your version number does not match this format, then the entire version number is treated as being the Qualifier.

for our purpose we should probably really deviate from the versioning scheme as otherwise we would be able to do only exact match based on whole string. anyway, i think this should probably also affect resolution of versioned packages from the cache. also, the deviation should be documented in the code.